drivers/spi/spi_mcux_flexcomm.c

```diff
@@ -218,6 +245,372 @@ static int spi_mcux_configure(const struct device *dev,
+
+	/* pass our client origin to the dma: data->dma_rx.channel */
+	ret = dma_config(data->dma_rx.dma_dev, data->dma_rx.channel,
+			&stream->dma_cfg);
+	/* the channel is the actual stream from 0 */
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* gives the request ID */
+	return dma_start(data->dma_rx.dma_dev, data->dma_rx.channel);
+}
+
+static int spi_mcux_dma_move_buffers(const struct device *dev, size_t len,
+			const struct spi_config *spi_cfg, bool last_packet)
+{
+	struct spi_mcux_data *data = dev->data;
+	int ret;
+
+	ret = spi_mcux_dma_rx_load(dev, data->ctx.rx_buf, len);
+
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = spi_mcux_dma_tx_load(dev, data->ctx.tx_buf, spi_cfg,
+							len, last_packet);
+
+	return ret;
+}
+
+static int wait_dma_rx_tx_done(const struct device *dev)
+{
+	struct spi_mcux_data *data = dev->data;
+	int ret = -1;
+
+	while (1) {
+		ret = spi_context_wait_for_completion(&data->ctx);
+		if (data->status_flags & SPI_MCUX_FLEXCOMM_DMA_ERROR_FLAG) {
+			return -EIO;
+		}
+
+		if ((data->status_flags & SPI_MCUX_FLEXCOMM_DMA_DONE_FLAG) ==
+			SPI_MCUX_FLEXCOMM_DMA_DONE_FLAG) {
+			return 0;
+		}
+	}
+}
+
+static int transceive_dma(const struct device *dev,
+		      const struct spi_config *spi_cfg,
+		      const struct spi_buf_set *tx_bufs,
+		      const struct spi_buf_set *rx_bufs,
+		      bool asynchronous,
+		      struct k_poll_signal *signal)
+{
+	const struct spi_mcux_config *config = dev->config;
+	struct spi_mcux_data *data = dev->data;
+	SPI_Type *base = config->base;
+	int ret;
+	uint32_t word_size;
+
+	spi_context_lock(&data->ctx, asynchronous, signal);
+
+	ret = spi_mcux_configure(dev, spi_cfg);
+	if (ret) {
+		goto out;
+	}
+
+	spi_context_buffers_setup(&data->ctx, tx_bufs, rx_bufs, 1);
+
+	spi_context_cs_control(&data->ctx, true);
+
+	word_size = SPI_WORD_SIZE_GET(spi_cfg->operation);
+
+	data->dma_rx.dma_cfg.dest_data_size = (word_size > 8) ?
+				(sizeof(uint16_t)) : (sizeof(uint8_t));
+	data->dma_tx.dma_cfg.dest_data_size = data->dma_rx.dma_cfg.dest_data_size;
+
+	while (data->ctx.rx_len > 0 || data->ctx.tx_len > 0) {
+		size_t dma_len;
+		bool last = false;
+
+		if (data->ctx.rx_len == 0) {
+			dma_len = data->ctx.tx_len;
+			last = true;
+		} else if (data->ctx.tx_len == 0) {
+			dma_len = data->ctx.rx_len;
+			last = true;
+		} else if (data->ctx.tx_len == data->ctx.rx_len) {
+			dma_len = data->ctx.rx_len;
+			last = true;
+		} else {
+			dma_len = MIN(data->ctx.tx_len, data->ctx.rx_len);
+			last = false;
+		}
+
+		data->status_flags = 0;
+
+		ret = spi_mcux_dma_move_buffers(dev, dma_len, spi_cfg, last);
+		if (ret != 0) {
+			break;
+		}
+
+		ret = wait_dma_rx_tx_done(dev);
+		if (ret != 0) {
+			break;
+		}
+
+		/* wait until TX FIFO is really empty */
+		while (0U == (base->FIFOSTAT & SPI_FIFOSTAT_TXEMPTY_MASK)) {
+		}
+
+		spi_context_update_tx(&data->ctx, 1, dma_len);
+		spi_context_update_rx(&data->ctx, 1, dma_len);
+	}
+
+	dma_stop(data->dma_rx.dma_dev, data->dma_rx.channel);
```

Comment:
dma_stop is not necessary, unless there is an error.

Reply:
This is end of the transfer, why are you suggesting we do not stop?

Reply:
this is not a must call, usually stop dma just to ensure it can be reoverred from error.

Reply:
In that case, should I check the value of `ret` before calling stop?

Reply:
no if there is an error, the error handle in dma driver shall already do this, so I suggest not call dam_stop here

Reply:
Got it, I will delete dma_stop